cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
If you want some indication that it the repo is collectible you could add a default CVMFS_GARBAGE_COLLECTION=false on those that are collectible, and leave it out completely on those that aren't.